### PR TITLE
Lower case url params in metric config

### LINKED ISSUE #317: [Frontend] Make all URLs lower case
@lilidworkin commented on [Fri Jan 13 2023](https://github.com/Recidiviz/recidiviz-data/issues/17863)

I think we might have addressed some of these (I seem to remember @terryttsai  doing some work on it), but I'm still seeing some upper case URLs, like https://justice-counts-web-qqec6jbn6a-uc.a.run.app/agency/153/settings/metric-config?system=SUPERVISION&metric=SUPERVISION_BUDGET. Is it easy/possible to make both the system and metric name lower case?

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

Make url params lower case for prettier URLs in the metric config

Screen Shot 2023-01-17 at 2 34 09 PM
Screen Shot 2023-01-17 at 2 34 18 PM

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #317

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: publisher/src/components/Settings/hooks.ts [hunk 1]
```diff
@@ -30,7 +30,9 @@ export const useSettingsSearchParams = (): [
   const { system, metric } = settingsSearchParams;
   const shouldReplace = !system && !metric;
   const setSettingsSearchParams = (params: SettingsSearchParams) => {
-    setSearchParams(params, { replace: shouldReplace });
+    setSearchParams(JSON.parse(JSON.stringify(params).toLowerCase()), {
```
Comment (lilidworkin) [inline publisher/src/components/Settings/hooks.ts:33]: Just OOC why do we need the JSON parse/stringify?

**Comment by mxosman** [comment]
LGTM!

File: publisher/src/components/Settings/hooks.ts (hunk 1 above)
Comment (terryttsai) [reply publisher/src/components/Settings/hooks.ts:33]: This was just the simplest way I know to transform every value in the object into lowercase!

Comment (mxosman) [reply publisher/src/components/Settings/hooks.ts:33]: Very clever!

**Review by mxosman — APPROVED** [review]

**Pull request merged**